elastic/kibana
# [Fleet] Use predefined id for default output

## Summary

Related to https://github.com/elastic/kibana/issues/111858#issuecomment-983953702 https://github.com/elastic/kibana/issues/118423

Use a predefined id for default id to make the setup more idempotent. Also added `overwrite` to the create call.


It does not have any impact on existing environment where the output will have an uuid.

## How to tests

I tested this modifying the setup to trigger race conditions 
```diff
diff --git a/x-pack/plugins/fleet/server/services/setup.ts b/x-pack/plugins/fleet/server/services/setup.ts
index 18c66e82674..ad35f643d3b 100644
--- a/x-pack/plugins/fleet/server/services/setup.ts
+++ b/x-pack/plugins/fleet/server/services/setup.ts
@@ -69,6 +69,21 @@ async function createSetupSideEffects(
     settingsService.settingsSetup(soClient),
   ]);

+  await Promise.all([
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+    outputService.ensureDefaultOutput(soClient),
+  ]);
+
   const defaultOutput = await outputService.ensureDefaultOutput(soClient);

   if (appContextService.getConfig()?.agentIdVerificationEnabled) {
 ```

Without that change we can get multiple default outputs see:
<img width="1680" alt="Screen Shot 2021-12-01 at 2 35 24 PM" src="https://user-images.githubusercontent.com/1336873/144301658-57afb3c8-4951-4028-9451-21e55d96a353.png">

With that change 
<img width="1631" alt="Screen Shot 2021-12-01 at 2 36 28 PM" src="https://user-images.githubusercontent.com/1336873/144301768-865cfa08-c1a3-4807-9ad1-51f3df7d6a1f.png">
